code/game/objects/items/storage/pill_packets.dm

```diff
@@ -2,71 +2,86 @@
 
 /obj/item/storage/pill_bottle/packet
 	name = "Pill packet"
-	desc = "Containts pills. Once you take them out they don't go back in."
-	greyscale_config = null //this line stops pill packets from turning into pill bottles
-	icon_state = "pill_packet"
+	desc = "Contains pills. Once you take them out they don't go back in."
+	icon_state = "packet_canister"
 	cant_hold = list(/obj/item/reagent_containers/pill) //Nada. Once you take the pills out. They don't come back in.
-	storage_slots = 4
+	storage_slots = 8
 	max_w_class = 0
-	max_storage_space = 4
+	max_storage_space = 8
+	greyscale_config = null //So that we get packets and not pill bottles.
+	///Color of the pips on top of the pill packet
+	var/pip_color = "#0066ff" //default color because I like this color
+	///the item left behind when this is used up
+	var/trash_item = /obj/item/reagent_containers/food/snacks/pillpacket
 	refill_types = null
 	refill_sound = null
 
+/obj/item/storage/pill_bottle/packet/remove_from_storage(obj/item/item, atom/new_location, mob/user)
+	. = ..()
+	if(!.)
+		return
+	update_icon()
+	if(!contents.len && !gc_destroyed)
+		var/turf/T = get_turf(src)
+		new trash_item(T)
+		qdel(src)
```

Comment:
we don't need to update icon if we're gonna qdel it anyway. So lets rewrite this with an early return with update icon instead.